js/ui/components/clubhouse.js

```diff
@@ -678,13 +590,17 @@ class ClubhouseNotificationSource extends NotificationDaemon.GtkNotificationDaem
 
 var Component = GObject.registerClass({
 }, class ClubhouseComponent extends SideComponent.SideComponent {
-    _init() {
-        super._init(ClubhouseIface, CLUBHOUSE_ID, CLUBHOUSE_DBUS_OBJ_PATH);
+    _init(clubhouseIface, clubhouseId, clubhousePath) {
+        this._clubhouseId = clubhouseId || CLUBHOUSE_ID;
+        this._clubhouseIface = clubhouseIface || CLUBHOUSE_IFACE;
+        this._clubhousePath = clubhousePath || CLUBHOUSE_DBUS_OBJ_PATH;
+
```

Comment:
Even with that solely isOldClubhouse argument, I think you can keep this attributes/variables. But decide the value depending on the isOldClubhouse argument.